public/apps/configuration/top-nav-menu.tsx

```diff
@@ -32,23 +32,25 @@ export const SecurityPluginTopNavMenu = (props: TopNavMenuProps) => {
     dataSourceManagement,
     setDataSource,
     selectedDataSource,
+    dataSourcePickerReadOnly,
   } = props;
   const { setHeaderActionMenu } = params;
-  const DataSourceMenu = dataSourceManagement?.ui.DataSourceMenu;
+  const DataSourceMenu = dataSourceManagement?.ui.getDataSourceMenu();
+
   const dataSourceEnabled = !!depsStart.dataSource?.dataSourceEnabled;
 
   return dataSourceEnabled ? (
     <DataSourceMenu
-      showDataSourceSelectable={dataSourceEnabled}
-      appName={PLUGIN_NAME}
-      savedObjects={coreStart.savedObjects.client}
       setMenuMountPoint={setHeaderActionMenu}
-      notifications={coreStart.notifications}
-      dataSourceCallBackFunc={setDataSource}
-      // Single select for now
-      selectedOption={[selectedDataSource]}
-      hideLocalCluster={false}
-      fullWidth={false}
+      componentType={dataSourcePickerReadOnly ? 'DataSourceView' : 'DataSourceSelectable'}
```

Comment:
Need to hard code these values, but opened an issue: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6271 in core to try and address this